sdk/textanalytics/azure-ai-textanalytics/src/test/java/com/azure/ai/textanalytics/ReferencePointerParseTest.java

```diff
@@ -0,0 +1,152 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.ai.textanalytics;
+
+import com.azure.ai.textanalytics.implementation.TextAnalyticsClientImpl;
+import com.azure.ai.textanalytics.implementation.models.DocumentSentiment;
+import com.azure.ai.textanalytics.implementation.models.SentenceOpinion;
+import com.azure.ai.textanalytics.implementation.models.SentenceSentiment;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.MockitoAnnotations;
+
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.Mockito.when;
+
+/**
+ * Unit tests for Aspect relation's reference parse
+ */
+public class ReferencePointerParseTest {
+
+    @Mock
```

Comment:
Just wanted to clarify that we are using mocks because the reference pointer is an internal function and can only be accessed with object mocking?

Reply:
I want to remove the dependency on TextAnalyticsClientImpl
```
private AnalyzeSentimentAsyncClient analyzeSentimentAsyncClient = new AnalyzeSentimentAsyncClient(textAnalyticsClientImpl);
```